### PR TITLE
Add option `dry-run` to do a dry run of config init command

### PR DESCRIPTION
**Pull request by namanpatel1112**
Resolves https://github.com/zowe/zowe-cli/issues/1037

### CODE REVIEW

File: packages/imperative/src/config/cmd/init/init.handler.ts [hunk 1]
```diff
@@ -88,6 +143,28 @@ export default class InitHandler implements ICommandHandler {
...
+     */
+    private async initForDryRun(config: Config, user: boolean): Promise<any> {
+        // Build the schema and write it to disk
+        const schema = ConfigSchema.buildSchema(ImperativeConfig.instance.loadedConfig.profiles);
+        config.setSchema(schema);
+
+        const opts: IConfigBuilderOpts = {};
+        if (!user) {
+            opts.populateProperties = true;
+            opts.getSecureValue = this.promptForProp.bind(this);
```
Comment (t1m0thyj) [inline packages/imperative/src/config/cmd/init/init.handler.ts:160]: No need to change anything here, just an observation 🙂 

The inclusion of prompting when `--dry-run` flag is passed struck me as odd at first when reviewing this PR, but it seems kind of necessary in order to show a complete diff of the changes.

File: packages/imperative/src/config/cmd/init/init.handler.ts [hunk 2]
```diff
@@ -42,18 +46,69 @@ export default class InitHandler implements ICommandHandler {
...
+            original = JSONC.stringify(originalProperties, null, ConfigConstants.INDENT);
+            dryRun = JSONC.stringify(dryRunProperties, null, ConfigConstants.INDENT);
+
+            let jsonDiff = diff(original, dryRun, {aAnnotation: "Removed",
+                bAnnotation: "Added",
+                aColor: TextUtils.chalk.red,
+                bColor: TextUtils.chalk.green});
+
+            if (stripAnsi(jsonDiff) === "Compared values have no visual difference.") {
+                jsonDiff = dryRun;
```
[the remark this answers is not in the document]
Comment (namanpatel1112) [reply packages/imperative/src/config/cmd/init/init.handler.ts:94]: @t1m0thyj @zFernand0 
Created issue for this to discuss it: https://github.com/zowe/imperative/issues/685

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

**Comment by namanpatel1112** [comment]
@zFernand0 Thanks for approving. I resolved the changelog conflict, but now the tests are failing.

**Comment by zFernand0** [comment]
> > but now the tests are failing.
> 
> That should be fine - we fixed those tests in another branch that we want to pull in after this PR is merged.

What Andrew said 😋 
Merging!

**Pull request merged**